src/SimplePie.php

```diff
@@ -1921,7 +1921,7 @@ protected function fetch_data(&$cache)
                             $this->status_code = $file->get_status_code();
                         } catch (HttpException $th) {
                             $this->check_modified = false;
-                            $this->status_code = 0;
+                            $this->status_code = $th->getCode();
```

Comment:
I think this makes sense for genuine HTTP errors but we sometimes promoted a random `Exception` to  `HTTPException`, even when there is no meaningful error code.

So I wonder if we should add a 1000 to the error code of promoted exceptions like this:

- IIUC, `File` will never throw an exception, unless some warning gets promoted to `Exception` by the library user’s custom error handler:
  https://github.com/simplepie/simplepie/blob/72b42935cb828ef691d17ca1e39fc0c9f480028b/src/HTTP/FileClient.php#L69
- HTTP Errors [do not throw exceptions](https://www.php-fig.org/psr/psr-18/#clientexceptioninterface) so this error code cannot be a HTTP status code:
  https://github.com/simplepie/simplepie/blob/72b42935cb828ef691d17ca1e39fc0c9f480028b/src/HTTP/Psr18Client.php#L117
- Definitely not a HTTP status code:
  https://github.com/simplepie/simplepie/blob/72b42935cb828ef691d17ca1e39fc0c9f480028b/src/HTTP/Psr18Client.php#L154
- https://github.com/simplepie/simplepie/blob/72b42935cb828ef691d17ca1e39fc0c9f480028b/src/HTTP/Psr18Client.php#L158

and then only change the status code when it does not:

```suggestion
                            $this->status_code = $th->getCode() < 1000 ? $th->getCode() : 0;
```

Though of course this will be easy to forget if we ad more `HTTPException` promotions.